libsodium-sys/build.rs

```diff
@@ -152,10 +152,9 @@ fn make_libsodium(target: &str, source_dir: &Path, install_dir: &Path) -> PathBu
     use std::{fs, process::Command, str};
 
     // Decide on CC, CFLAGS and the --host configure argument
-    let build = cc::Build::new();
-    let mut compiler = build.get_compiler().path().to_str().unwrap().to_string();
-    let mut cflags = env::var("CFLAGS").unwrap_or(String::default());
-    cflags += " -O2";
+    let build_compiler = cc::Build::new().get_compiler();
+    let mut compiler = build_compiler.path().to_str().unwrap().to_string();
+    let mut cflags = build_compiler.cflags_env().into_string().unwrap();
```

Comment:
Why .unwrap() instead of .unwrap_or(..)?

Reply:
`.unwrap()` asserts the validity of the conversion from `OsString` to `String` similar to `path().to_str().unwrap()` above. Generally, the build script seems to assume that paths and OS strings are valid UTF-8 and uses `unwrap` where this needs to be asserted. So I tried to keep things consistent and also to avoid silently dropping user-supplied flags if there are not valid UTF-8.